raid.go

```diff
@@ -91,8 +94,24 @@ func (ar RaidArrays) Measurements() MeasurementsMap {
 		results[raid.Name+".state"] = raid.State
 		results[raid.Name+".type"] = raid.Type
 
-		if raid.State == RaidStatusDegraded {
-			results[raid.Name+".failed"] = strings.Join(raid.GetFailedPhysicalDevices(), "; ")
+		failedDevices, missingCount := raid.GetFailedAndMissingPhysicalDevices()
+
+		if len(failedDevices) > 0 || missingCount > 0 {
+			results[raid.Name+".degraded"] = 1
+
+			results[raid.Name+".physicaldevice.missing"] = missingCount
+
+			for _, failedDevice := range failedDevices {
```

Comment:
Sort of ok in this scenario but I suggest you try to keep the branches inside `if/else` shorter.
Don't nest `if/else` and loops. This makes the code hard to read and functions can grow very quickly.
Try to extract the code within the branches into functions. Makes it easier to test, read and refactor.

Reply:
In this case, we have the combined data source for all raid arrays. If we will split the outer loop(arrays) and inner loops it will require to reparse the original data or store it in-the-middle